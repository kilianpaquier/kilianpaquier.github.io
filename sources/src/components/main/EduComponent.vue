<template>
  <v-container fluid>
    <!-- Education title -->
    <h2
      class="font-weight-bold mb-3"
      v-text="$vuetify.lang.t(`$vuetify.educations-title`)"
    />

    <!-- Education timeline -->
    <v-timeline dense clipped>
      <v-timeline-item v-for="(edu, i) in educations" :key="i" large>
        <!-- Avatar part of the timeline item -->
        <template v-slot:icon>
          <v-avatar class="bg-white">
            <v-img
              :src="`/img/educations/${edu.img}`"
              class="bg-white"
              eager
            />
          </v-avatar>
        </template>

        <!-- Card part of the timeline item -->
        <v-card>
          <!-- Title, subtitle and text part -->
          <v-card-title v-text="edu.title" />
          <v-card-subtitle
            class="pb-0"
            v-text="$vuetify.lang.t(`$vuetify.${edu.dates}`)"
          />
          <v-card-text
            :class="{ 'py-0': edu.caption, 'pt-0': !edu.caption }"
            v-text="$vuetify.lang.t(`$vuetify.${edu.value}`)"
          />

          <!-- Captions if needed -->
          <v-card-actions class="caption" v-if="edu.caption">
            <v-spacer></v-spacer>
            <span v-text="edu.caption" />
          </v-card-actions>
        </v-card>
      </v-timeline-item>
    </v-timeline>
  </v-container>
</template>

<script lang="ts">
import { Education } from "@/model/entities";
import Vue from "vue";
import Component from "vue-class-component";

@Component
export default class EduComponent extends Vue {
  educations: Education[] = [
      {
          title: "PSM I",
          dates: "educations.psm-one.dates",
          img: "psmi.svg",
          value: "educations.psm-one.title",
      },
      {
          title: "UQAC¹",
          dates: "educations.uqac.dates",
          img: "uqac.jpg",
          value: "educations.uqac.title",
          caption: "¹ Université du Québec à Chicoutimi",
      },
      {
          title: "Polytech Tours",
          dates: "educations.polytech.dates",
          img: "polytechtours.jpg",
          value: "educations.polytech.title",
      },
  ];
}
</script>
