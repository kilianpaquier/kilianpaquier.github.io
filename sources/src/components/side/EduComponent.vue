<template>
    <k-row title="educations_title">
        <!-- Education timeline -->
        <v-timeline dense clipped>
            <v-timeline-item v-for="(education, index) in educations" :key="'edu' + index" large>
                <!-- Avatar part of the timeline item -->
                <template v-slot:icon>
                    <v-avatar class="bg-white">
                        <v-img v-bind:src="education.img" class="bg-white" eager/>
                    </v-avatar>
                </template>

                <!-- Card part of the timeline item -->
                <v-card>
                    <v-list>
                        <k-list-item v-bind:title="education.title" v-bind:lines="3">
                            <v-list-item-subtitle>
                                {{ $t(education.dates) }}
                            </v-list-item-subtitle>

                            <v-list-item-subtitle>
                                {{ $t(education.diploma) }}
                            </v-list-item-subtitle>

                            <v-list-item-subtitle class="pt-2" v-if="education.skills">
                                <v-row no-gutters>
                                    <v-col cols="auto" class="pe-2 py-1" v-for="(skill, index) in education.skills" v-bind:key="'exp_skill' + index">
                                        <k-skill v-bind:icon="skill.icon" v-bind:title="skill.name"></k-skill>
                                    </v-col>
                                </v-row>
                            </v-list-item-subtitle>
                        </k-list-item>
                    </v-list>

                    <!-- Captions if needed -->
                    <v-card-actions class="caption" v-if="education.caption">
                        <v-spacer></v-spacer>
                        <span v-text="education.caption"/>
                    </v-card-actions>
                </v-card>
            </v-timeline-item>
        </v-timeline>
    </k-row>
</template>

<script lang="ts">
import {defineComponent} from "vue";
import KRow from "../common/KRow.vue";
import KSkill from "../common/KSkill.vue";
import psmi from "../../assets/educations/psmi.svg";
import uqac from "../../assets/educations/uqac.jpg";
import polytech from "../../assets/educations/polytechtours.jpg";
import {
    bootstrap,
    c,
    cplusplus,
    hibernate,
    java,
    javascript,
    php,
    python,
    spring,
    qt, jquery, html5, perforce, unity, csharp, dotnet, android, moleculer
} from "../../plugins/devicon";
import KListItem from "../common/KListItem.vue";

export default defineComponent({
    name: "EduComponent",
    components: {
        KListItem,
        KSkill,
        KRow
    },

    data() {
        return {
            educations: [
                {
                    title: "PSM I",
                    dates: "educations[0].dates",
                    img: psmi,
                    diploma: "educations[0].diploma",
                },
                {
                    title: "UQAC¹",
                    dates: "educations[1].dates",
                    img: uqac,
                    diploma: "educations[1].diploma",
                    caption: "¹ Université du Québec à Chicoutimi",
                    skills: [
                        { icon: java, name: "Java" },
                        { icon: spring, name: "Spring" },
                        { icon: hibernate, name: "Hibernate" },
                        { icon: android, name: "Android" },

                        { icon: javascript, name: "Javascript" },
                        { icon: moleculer, name: "Moleculer" },

                        { icon: html5, name: "HTML" },
                        { icon: bootstrap, name: "Bootstrap" },

                        { icon: csharp, name: "C#" },
                        { icon: dotnet, name: ".NET Framework" },

                        { icon: perforce, name: "Perforce" },
                        { icon: unity, name: "Unity" },
                    ]
                },
                {
                    title: "Polytech Tours",
                    dates: "educations[2].dates",
                    img: polytech,
                    diploma: "educations[2].diploma",
                    skills: [
                        { icon: java, name: "Java" },
                        { icon: hibernate, name: "Hibernate" },

                        { icon: html5, name: "HTML" },
                        { icon: bootstrap, name: "Bootstrap" },

                        { icon: javascript, name: "Javascript" },
                        { icon: jquery, name: "JQuery" },

                        { icon: python, name: "Python" },
                        { icon: php, name: "PHP" },

                        { icon: c, name: "C" },
                        { icon: cplusplus, name: "C++" },
                        { icon: qt, name: "Qt" },
                    ],
                },
            ]
        }
    }
});
</script>

<style scoped>

</style>
