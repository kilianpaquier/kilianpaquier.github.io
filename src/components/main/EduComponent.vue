<template>
    <v-row :no-gutters="true">
        <v-col cols="12">
            <h2 class="mb-3">{{ $t("educations_title") }}</h2>
        </v-col>
        <v-col cols="12">
            <v-timeline side="end" density="comfortable">
                <v-timeline-item v-for="(education, index) in educations" :key="'education_' + index" :fill-dot="true" size="large" width="100%">
                    <template #icon>
                        <v-avatar class="h-100 w-100">
                            <v-img :cover="true" :eager="true" :sizes="sizes" :src="education.src" :srcset="education.srcset" class="w-100 h-100" />
                        </v-avatar>
                    </template>

                    <v-card :title="education.title">
                        <v-card-subtitle>{{ $t(education.dates) }}</v-card-subtitle>
                        <v-card-subtitle>{{ $t(education.diploma) }}</v-card-subtitle>

                        <v-card-text :class="{ 'pb-0': !education.skills || education.caption }" class="pt-3">
                            <v-row v-if="education.skills" :no-gutters="true">
                                <v-col v-for="(skill, index) in education.skills" :key="'experience_skill' + index" :class="{ 'me-2': index < education.skills.length - 1 }" class="my-1" cols="auto">
                                    <v-tooltip :text="skill.name" location="bottom">
                                        <template #activator="{props}">
                                            <component :is="skill.icon" class="mx-auto no-outline" v-bind="props" width="25" />
                                        </template>
                                    </v-tooltip>
                                </v-col>
                            </v-row>
                        </v-card-text>

                        <v-card-actions v-if="education.caption" class="text-caption">
                            <v-spacer />
                            <span>
                                {{ education.caption }}
                            </span>
                        </v-card-actions>
                    </v-card>
                </v-timeline-item>
            </v-timeline>
        </v-col>
    </v-row>
</template>

<script lang="ts">
import {defineComponent} from "vue";
import educations from "@/data/educations";

export default defineComponent({
    name: "EduComponent",

    setup: () => ({
        sizes: "10vw",
        educations
    })
});
</script>