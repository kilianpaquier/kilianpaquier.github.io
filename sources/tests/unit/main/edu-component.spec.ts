import { createLocalVue, mount, Wrapper } from "@vue/test-utils";
import EduComponent from "@/components/main/EduComponent.vue";
import Vue from "vue";
import Vuetify from "vuetify";

Vue.use(Vuetify);
Vue.config.silent = true;
Vuetify.config.silent = true;

describe("EduComponent.vue", () => {
    const localVue = createLocalVue();

    let vuetify: Vuetify;
    let wrapper: Wrapper<EduComponent>;

    beforeEach(() => {
        vuetify = new Vuetify();

        wrapper = mount(EduComponent, {
            localVue,
            vuetify,
        });
    });

    it("should create EduComponent", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});
