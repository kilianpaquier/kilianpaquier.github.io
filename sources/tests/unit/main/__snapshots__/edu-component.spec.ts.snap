// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EduComponent.vue should create EduComponent 1`] = `
<div class="container container--fluid">
  <h2 class="font-weight-bold mb-3">$vuetify.educations-title</h2>
  <div class="v-timeline v-timeline--dense theme--light" clipped="">
    <div class="v-timeline-item theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">PSM I</div>
          <div class="v-card__subtitle pb-0">$vuetify.educations.psm-one.dates</div>
          <div class="v-card__text pt-0">$vuetify.educations.psm-one.title</div>
          <!---->
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--large">
          <div class="v-timeline-item__inner-dot primary">
            <div class="v-avatar bg-white" style="height: 48px; min-width: 48px; width: 48px;">
              <div class="v-image v-responsive bg-white theme--light">
                <div class="v-responsive__content"></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="v-timeline-item theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">UQAC¹</div>
          <div class="v-card__subtitle pb-0">$vuetify.educations.uqac.dates</div>
          <div class="v-card__text py-0">$vuetify.educations.uqac.title</div>
          <div class="v-card__actions caption">
            <div class="spacer"></div> <span>¹ Université du Québec à Chicoutimi</span>
          </div>
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--large">
          <div class="v-timeline-item__inner-dot primary">
            <div class="v-avatar bg-white" style="height: 48px; min-width: 48px; width: 48px;">
              <div class="v-image v-responsive bg-white theme--light">
                <div class="v-responsive__content"></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="v-timeline-item theme--light">
      <div class="v-timeline-item__body">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title">Polytech Tours</div>
          <div class="v-card__subtitle pb-0">$vuetify.educations.polytech.dates</div>
          <div class="v-card__text pt-0">$vuetify.educations.polytech.title</div>
          <!---->
        </div>
      </div>
      <div class="v-timeline-item__divider">
        <div class="v-timeline-item__dot v-timeline-item__dot--large">
          <div class="v-timeline-item__inner-dot primary">
            <div class="v-avatar bg-white" style="height: 48px; min-width: 48px; width: 48px;">
              <div class="v-image v-responsive bg-white theme--light">
                <div class="v-responsive__content"></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
